sdk/src/main/java/com/hedera/hashgraph/sdk/Executable.java

```diff
@@ -156,12 +162,72 @@ public SdkRequestT setNodeAccountIds(List<AccountId> nodeAccountIds) {
         if (minBackoff == null) {
             minBackoff = client.getMinBackoff();
         }
+    }
 
-        return onExecuteAsync(client).thenCompose((v) -> {
-            if (nodeAccountIds.isEmpty()) {
-                throw new IllegalStateException("Request node account IDs were not set before executing");
+    void delay(long delay) {
+        try {
+            Thread.sleep(delay);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public O execute(Client client) throws TimeoutException, PrecheckStatusException {
```

Comment:
That is fair, but _I think_ this change is tested by a test you wrote awhile ago https://github.com/hashgraph/hedera-sdk-java/blob/0407027cb0bd665da79ae6ae33f89cc3a9ad24ce/sdk/src/test/java/com/hedera/hashgraph/sdk/MockingTest.java#L24 This tests calls the `AccountBalanceQuery.execute()` method which means it would call the overridden method in this PR. Perhaps we should add a test for the `executeAsync()` since it's been changed.

Reply:
I didn't add `MockingTest`. I think Sean did.

But I don't think one test provides good coverage. I see you also added a test for maxattempts, so that's good. But I would like to see the tests ran in an IDE with coverage turned on. It will help you visualize which lines are green with good coverage and those that are not. 

The test you mention provides good coverage for `shouldRetryExceptionally()`, but not much else. Should also test the different outputs of `shouldRetry()` along with the different failures of get cost query.